### PR TITLE
fix typo and upgrade dependencies

### PR DESCRIPTION
**Pull request by zhaonian**
TESTED=app builds and runs as expected

### CODE REVIEW

**Comment by owahltinez** [comment]
This does not compile for me:
```
app/src/main/java	
com/android/example/cameraxbasic/fragments/CameraFragment.kt	
Object is not abstract and does not implement abstract member public abstract fun onError(@NonNull p0: ImageCapture.ImageCaptureError, @NonNull p1: String, @Nullable p2: Throwable?): Unit defined in androidx.camera.core.ImageCapture.OnImageSavedListener	
'onError' overrides nothing	
Unresolved reference: UseCaseError	
com/android/example/cameraxbasic/fragments/GalleryFragment.kt	
Cannot inline bytecode built with JVM target 1.8 into bytecode that is being built with JVM target 1.6. Please specify proper '-jvm-target' option	
```

**Comment by zhaonian** [comment]
can't believe I missed it. Just updated the code. thx!

**Comment by owahltinez** [comment]
Application now crashes, at least on Pixel 3 XL device:
```
E/AndroidRuntime: FATAL EXCEPTION: LuminosityAnalysis
    Process: com.android.example.cameraxbasic, PID: 18995
    java.lang.IllegalStateException: Not in application's main thread
        at androidx.core.util.Preconditions.checkState(Preconditions.java:96)
        at androidx.camera.core.impl.utils.Threads.checkMainThread(Threads.java:52)
        at androidx.camera.core.ImageAnalysis.getAnalyzer(ImageAnalysis.java:180)
        at com.android.example.cameraxbasic.fragments.CameraFragment$bindCameraUseCases$1$1.invoke(CameraFragment.kt:314)
        at com.android.example.cameraxbasic.fragments.CameraFragment$bindCameraUseCases$1$1.invoke(CameraFragment.kt:89)
        at com.android.example.cameraxbasic.fragments.CameraFragment$LuminosityAnalyzer.analyze(CameraFragment.kt:468)
        at androidx.camera.core.ImageAnalysisAbstractAnalyzer.analyzeImage(ImageAnalysisAbstractAnalyzer.java:57)
        at androidx.camera.core.ImageAnalysisNonBlockingAnalyzer$1.run(ImageAnalysisNonBlockingAnalyzer.java:135)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)

```

I think this is related to some internal changes with thread management for image analyzer. Please test on a couple of real devices, in addition to the emulator.

**Comment by zhaonian** [comment]
interesting. it looks like the callBackHandler for Analyzer has to be on the Main Thread now for some reason :/

File: CameraXBasic/app/build.gradle [hunk 1]
```diff
@@ -36,6 +36,10 @@ android {
         targetCompatibility rootProject.ext.java_version
     }
 
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8
```
Comment (owahltinez) [inline CameraXBasic/app/build.gradle]: Can you please use the versioned varaible? I.e. "$rootProject.ext.java_version"

**Comment by owahltinez** [comment]
I don't think the issue is with the thread the analyzer runs in, but rather when the analyzer is accessed later to log the results. We can keep this for the time being and add the background thread later after we investigate further.

Please address comments and rebase with the tip of master, then we can merge this.

File: CameraXBasic/app/build.gradle (hunk 1 above)
Comment (zhaonian) [reply CameraXBasic/app/build.gradle]: Done.

**Comment by zhaonian** [comment]
Sounds good! 
rebased/squashed all commits onto master.
thx!

**Pull request merged**